### PR TITLE
support to set the expanded panels of Explorer

### LINKED ISSUE #726: feat: support to set the expanded panels of Explorer
### Description

The default Explorer Panels are closed, and Molecule hasn't provided an API that could control it.  eg:

image

### Solution

Provide an API like this:

```ts
molecule.explorer.setExpandedPanels([panelId, panelId]);
```

### PR DESCRIPTION
**Pull request by mumiao**
## Description

support to set the expanded panels of Explorer

Fixes #726 

## Changes

Please describe the changes, help the reviewer to read your code.

-   support to set the expanded panels of Explorer
-   Collapse component support expand activeKeys
-   add  Explorer setExpandedPanels testcase
-   add  Collapse component activePanelKeys attribute testcase

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/collapse/index.tsx [hunk 1]
```diff
@@ -79,6 +83,16 @@ export function Collapse({
     const adjustedSize = useRef<number[]>([]);
     const first = useRef(true);
 
+    const useCampare = (value: UniqueId[], compare: Function) => {
```
Comment (mortalYoung) [inline src/components/collapse/index.tsx:86]: 自定义的 hook 放到函数体的外面呗

File: src/services/workbench/explorer/explorerService.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ export interface IExplorerService extends Component<IExplorer> {
      * Update the panels data, as well as modify toolbar data
      */
     updatePanel(data: Partial<IExplorerPanelItem>): void;
+    setExpandedPanels(activePanelKeys: UniqueId[]): void;
```
Comment (mortalYoung) [inline src/services/workbench/explorer/explorerService.ts:26]: services 的方法记得注释

File: src/components/collapse/index.tsx [hunk 3]
```diff
@@ -241,6 +255,12 @@ export function Collapse({
         first.current = false;
     }, [activePanelKeys, data]);
 
+    const compareActivePanelKeys = useCampare(controlActivePanelKeys, isEqual);
+
+    useLayoutEffect(() => {
+        setActivePanelKeys(compareActivePanelKeys);
+    }, [compareActivePanelKeys]);
```
Comment (mortalYoung) [inline src/components/collapse/index.tsx:262]: 这一段没太看懂，直接把 controlActivePanelKeys 丢到 deps 里然后 set 不阔以吗 0.0

**Review by mortalYoung — CHANGES_REQUESTED** [review]

File: src/components/collapse/index.tsx (hunk 3 above)
Comment (mumiao) [reply src/components/collapse/index.tsx:262]: 不可以, controlActivePanelKeys 是引用对象, 导致 render loop 了

**Review by mortalYoung — APPROVED** [review]
+1

**Review by wewoor — APPROVED** [review]
+1

**Pull request merged**